File: node/Foo.test.js
const Foo = require('./Foo')

describe('Foo tests', () => {
  it('Returns Foo', () => {
    expect(Foo.getFoo()).toMatch(/Foo/)
  })
})

describe('Fibonacci tests', () => {
  it('computes Fibonacci correctly', () => {
    expect(Foo.fibonacci(1)).toEqual(1)
    expect(Foo.fibonacci(2)).toEqual(2)
    expect(Foo.fibonacci(3)).toEqual(3)
    expect(Foo.fibonacci(4)).toEqual(5)
  })
  it('Numbers less than 0 return 1', () => {
    expect(Foo.fibonacci(0)).toEqual(1)
    expect(Foo.fibonacci(-1)).toEqual(1)
  })
})

describe('Person class tests', () => {
  it('Full name returns correct value', () => {
    const p = new Foo.Person ('John', 'Smith')
    expect(p.fullName()).toEqual('John Smith')
  })
})

File: java/FooTest.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;

public class FooTest
{
	@Test
	void testGetFoo() {
		assertEquals("Foo!", Foo.getFoo(), "Must expect \"Foo!\"");
	}

	@Test
	void testFibonacci() {
		assertEquals(1, Foo.fibonacci(1), "fibonacci(0) != 1");
		assertEquals(1, Foo.fibonacci(1), "fibonacci(1) != 1");
		assertEquals(2, Foo.fibonacci(2), "fibonacci(2) != 2");
		assertEquals(3, Foo.fibonacci(3), "fibonacci(3) != 3");
		assertEquals(5, Foo.fibonacci(4), "fibonacci(4) != 5");
	}

	@Test
	void testPersonClass() {
		Person p = new Person("John", "Smith");
		assertEquals("John Smith", p.fullName());
	}

	@Test
	void testFail() {
		assertEquals(1, 2, "Fix Me!");
	}
}